search-processors/src/test/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQARequestProcessorTests.java

```diff
@@ -18,32 +18,71 @@
 package org.opensearch.searchpipelines.questionanswering.generative;
 
 import org.opensearch.action.search.SearchRequest;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.search.pipeline.Processor;
 import org.opensearch.search.pipeline.SearchRequestProcessor;
 import org.opensearch.test.OpenSearchTestCase;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.BooleanSupplier;
+
+import static org.mockito.Mockito.mock;
 
 public class GenerativeQARequestProcessorTests extends OpenSearchTestCase {
 
+    private BooleanSupplier alwaysOn = () -> true;
+
     public void testProcessorFactory() throws Exception {
 
         Map<String, Object> config = new HashMap<>();
         config.put("model_id", "foo");
         SearchRequestProcessor processor =
-            new GenerativeQARequestProcessor.Factory().create(null, "tag", "desc", true, config, null);
+            new GenerativeQARequestProcessor.Factory(alwaysOn).create(null, "tag", "desc", true, config, null);
         assertTrue(processor instanceof GenerativeQARequestProcessor);
     }
 
     public void testProcessRequest() throws Exception {
-        GenerativeQARequestProcessor processor = new GenerativeQARequestProcessor("tag", "desc", false, "foo");
+        GenerativeQARequestProcessor processor = new GenerativeQARequestProcessor("tag", "desc", false, "foo", alwaysOn);
         SearchRequest request = new SearchRequest();
         SearchRequest processed = processor.processRequest(request);
         assertEquals(request, processed);
     }
 
     public void testGetType() {
-        GenerativeQARequestProcessor processor = new GenerativeQARequestProcessor("tag", "desc", false, "foo");
+        GenerativeQARequestProcessor processor = new GenerativeQARequestProcessor("tag", "desc", false, "foo", alwaysOn);
         assertEquals(GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE, processor.getType());
     }
+
+    public void testProcessorFactoryFeatureFlagDisabled() throws Exception {
+        Map<String, Object> config = new HashMap<>();
+        config.put("model_id", "foo");
+        Processor.Factory factory = new GenerativeQARequestProcessor.Factory(()->false);
+        try {
+            factory.create(null, "tag", "desc", true, config, null);
+        } catch (MLException e) {
+            assertEquals(GenerativeQAProcessorConstants.FEATURE_NOT_ENABLED_ERROR_MSG, e.getMessage());
+        }
+    }
```

Comment:
It is not a correct way to test throwing exception, as the case would pass if factory.create() throw nothing which is a wrong behavior.
You can use ExpectedException to test exception throwing, it is easy to find an example in the existing code.

Reply:
I foudn it.